llvm/lib/Transforms/IPO/ArgumentPromotion.cpp

```diff
@@ -611,6 +639,33 @@ static bool findArgParts(Argument *Arg, const DataLayout &DL, AAResults &AAR,
       // unknown users
     }
 
+    auto *CB = dyn_cast<CallBase>(V);
+    Value *PtrArg = cast<Value>(U);
```

Comment:
Writing this as `cast<Value>(U)` works, but is pretty weird. `U.get()` would be better.